opentelemetry/proto/trace/v1/trace.proto

```diff
@@ -329,5 +343,26 @@ enum SpanFlags {
   // Bits 0-7 are used for trace flags.
   SPAN_FLAGS_TRACE_FLAGS_MASK = 0x000000FF;
 
-  // Bits 8-31 are reserved for future use.
+  // Bits 8 and 9 are used to indicate that the parent span or link span is remote.
+  // See SpanFlagsContextIsRemote for the valid values.
+  SPAN_FLAGS_CONTEXT_IS_REMOTE_MASK = 0x00000300;
+
+  // Bits 10-31 are reserved for future use.
+}
+
+// SpanFlagsContextIsRemote represents constants used to interpret
+// bits 8 and 9 of the Spans.flags or Link.flags field. These two bits
+// represent whether the parent or link is remote. Bit 8 represents whether
+// the field has been set and bit 9 represents whether the parent or link is
+// remote.
+// Once the bits have been read using SPAN_FLAGS_CONTEXT_IS_REMOTE_MASK,
+// the following constants can be used to interpret them.
+enum SpanFlagsContextIsRemote {
```

Comment:
Do we need this extra enum? We agreed to offer only masks values, this is getting messier to have all possible values here and again my comment about mis-using them gets back to this.

Reply:
@bogdandrutu We agreed that 0 values for enums should not be used since it is easy to misuse them. This was the PR we merged after that discussion: https://github.com/open-telemetry/opentelemetry-proto/pull/461

However, I don't think we said that other (non-zero) values of enums are also discouraged.

I am open to discussing but I think it useful to have this enum. The enum codifies what otherwise would be just comments. We need that information in some form otherwise unclear what are the values for the 2 bits.

Reply:
Can we postpone adding this until we agree on that? Also this way we don't block the main functionality, and can always add the helper later.

Reply:
Sounds good to me. We can remove the enum for now and discuss it separately.

Reply:
To make progress on this: @estolfo can you please remove the enum for now? We can then create a follow-up PR that adds the enum and discuss it separately.

Reply:
@tigrannajaryan I just watched the meeting recording. Thanks for pinging me on this. I've removed the enum and updated the comment to include a brief description.